tests/testAfterburners.py

```diff
@@ -0,0 +1,181 @@
+    results out to the table
+    '''
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.DEFAULT_AFTERBURNER
+
+    @property
+    def PlugType(self):
+        return 'multi'
+
+    def __init__(self, config, name, schema, metadata):
+        AB.AfterburnerPlugin.__init__(self, config, name, schema, metadata)
+        self.failKey = schema.addField(name+"_fail", type="Flag", doc="Failure flag")
+        self.cubeKey = schema.addField(name+"_cube", type="D", doc="Cube of input catalog")
+
+    def burn(self, catalog):
+        for rec in catalog:
+            value = rec.get("start")
+            rec.set(self.cubeKey, value**3)
+
+    def fail(self, catalog, error=None):
+        for rec in catalog:
+            value = rec.set(self.failKey, True)
+
+@register("dependentAfterburner")
+class DependentAb(AB.AfterburnerPlugin):
+    '''
+    Afterburner plugin to test the runlevel resolution. This plugin takes in single records, reads a value
+    from a previous plugin computes a square root, and writes the results to the table.
+    '''
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.DEFAULT_AFTERBURNER + 1
+
+    def __init__(self, config, name, schema, metadata):
+        AB.AfterburnerPlugin.__init__(self, config, name, schema, metadata)
+        self.failKey = schema.addField(name+"_fail", type="Flag", doc="Failure flag")
+        self.sqrtKey = schema.addField(name+"_sqrt", type="D", doc="Square root of singleRecord afterburner")
+
+    def burn(self, measRecord):
+        value = measRecord.get("singleRecordAfterburner_square")
+        measRecord.set(self.sqrtKey, value**0.5)
+
+    def fail(self, measRecord, error=None):
+        measRecord.set(self.failKey, True)
+
+class AfterburnerTest(unittest.TestCase):
+    '''
+    Test the after burner framework, running only the plugins defined above
+    '''
+    def setUp(self):
+        # Create a schema object, and populate it with a field to simulate results from measurements on an
+        # image
+        schema = afwTable.SourceTable.makeMinimalSchema()
+        startKey = schema.addField("start", type="D")
+        # Instantiate a config object adding each of the above plugins, and use it to create a task
+        abConfig = AB.AfterburnerConfig()
+        abConfig.plugins.names = ["afterburnerFail", "singleRecordAfterburner",
+                "multiRecordAfterburner", "dependentAfterburner"]
+        abTask = AB.AfterburnerTask(schema=schema, config=abConfig)
+        # Create a catalog with five sources as input to the task
+        self.catalog = afwTable.SourceCatalog(schema)
+        self.numObjects = 5
+        for i in range(self.numObjects):
+            rec = self.catalog.addNew()
+            rec.set("start", float(i+1))
+        # Run the afterburner task, outputs will be checked in test methods
+        abTask.run(self.catalog)
+
```

Comment:
I don't see a `tearDown` function, which is usually necessary for the `MemoryTestCase` to pass when you hold various afw classes in the `TestCase` instance.  Maybe that's because nothing in `SourceCatalog` uses `Citizen`?